sdk/include/reactos/unaligned.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * PROJECT:     ReactOS SDK
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Unaligened access helper functions
+ * COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(_M_IX86) || defined(_M_AMD64)
+#define _UNALIGNED_ACCESS_ALLOWED 1
+#else // defined(_M_IX86) || defined(_M_AMD64)
+#define _UNALIGNED_ACCESS_ALLOWED 0
+#endif // defined(_M_IX86) || defined(_M_AMD64)
+
+#if _UNALIGNED_ACCESS_ALLOWED
+
+__forceinline
+unsigned short
+ReadUnalignedU16(const unsigned short* p)
+{
+    return *p;
+}
+
+__forceinline
+unsigned long
+ReadUnalignedU32(const unsigned long* p)
+{
+    return *p;
+}
+
+__forceinline
+unsigned long long
+ReadUnalignedU64(const unsigned long long* p)
+{
+    return *p;
+}
+
+__forceinline
+void
+WriteUnalignedU16(unsigned short* p, unsigned short val)
+{
+    *p = val;
+}
+
+__forceinline
+void
+WriteUnalignedU32(unsigned long* p, unsigned long val)
+{
+    *p = val;
+}
+
+__forceinline
+void
+WriteUnalignedU64(unsigned long long* p, unsigned long long val)
+{
+    *p = val;
+}
+
+#else // !UNALIGNED_ACCESS_ALLOWED
+
+__forceinline
+unsigned short
+ReadUnalignedU16(const unsigned short* p)
```

Comment:
Nice, this can be used as a replacement to `‎DcRetrieveWord` https://github.com/reactos/reactos/blob/master/drivers/network/dd/dc21x4/util.h#L44

Reply:
It could. Not sure, if it's worth changing though. It would break build with Win DDK (unless the file was included).